frontend/src/components/TemplateFlow.js

```diff
@@ -219,7 +236,8 @@ const TemplateFlow = (props) => {
     setIsSubmitting(true)
     submitRequest(actionIndex, file)
     .then(response => {
-      setSubmitResult({ valid: true })
+      setSubmitResult({ valid: true });
+      fetchInitialData();
```

Comment:
Not sure if this is the behavior you expect, but the `fetchInitialData()` promise isn't awaited on here.